src/lib/Footer.svelte

```diff
@@ -4,9 +4,30 @@
   import { faPatreon } from '@fortawesome/free-brands-svg-icons'
   import { faYoutube } from '@fortawesome/free-brands-svg-icons'
   import { faXTwitter } from '@fortawesome/free-brands-svg-icons'
+
+  import { showNSFW } from './storeNsfwPreference'
+  import { toggleVisibility } from './storeNsfwPreference'
+
+  function toggleNSFW() {
+    showNSFW.update((value) => !value)
+  }
 </script>
 
 <footer>
+  {#if $toggleVisibility}
+    <div class="nsfw-toggle-container">
+      <div class="nsfw-component" title="Toggle blur for NSFW covers if present">
+        <span class="toggle-title">NSFW Filter:</span>
+        <div class="toggle-wrapper" on:click={toggleNSFW}>
+          <div class="toggle-pill" class:active={$showNSFW}>
+            <span class="toggle-status"
+              >{#if $showNSFW}ON{:else}OFF{/if}</span
```

Comment:
Not necessarily better alternative (although I am not sure whether it works with a store):

`{$showNSFW ? 'ON' : 'OFF'}`